package Exceptions.Throwing_Exceptions;

//

// https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html


/*                                  Exception object


    ○ When an error occurs within a method,

                   * the method CREATES AN OBJECT

                   * and HANDS IT OFF to the runtime system.

           The object, called an Exception Object,

                    * CONTAINS information about the error,

                                  - including its type

                                  - and the state of the program

                          when the error occurred.


                                    Throwing an exception.


    ○ Creating

                   * an Exception object

                   * and handing it to the runtime system

           is called throwing an exception.


    ○ After a method throws an exception,

                  * the runtime system attempts to find something to handle it.

          The set of possible "something's" to handle the exception is

                  * the ordered list of methods that had been called to get to the method where the error occurred.

          The list of methods is known as the

                   * call stack    -- see the  exceptions-callstack.gif




 */


public class Throwing_Exceptions_ {




}
